python/dgl/nn/pytorch/sparse_emb.py

```diff
@@ -376,6 +376,36 @@ def all_set_embedding(self, values):
         if th.distributed.is_initialized():
             th.distributed.barrier()
 
+    def _all_get_tensor(self, shared_name, tensor, shape):
+        """A helper function to get model-parallel tensors."""
+        # create a shared memory tensor
+        if self._rank == 0:
+            # root process creates shared memory
+            val = create_shared_mem_array(
+                shared_name,
+                shape,
+                tensor.dtype,
+            )
+            self._store.set(shared_name, shared_name)
+        else:
+            self._store.wait([shared_name])
+            val = get_shared_mem_array(
+                shared_name,
+                shape,
+                tensor.dtype,
+            )
+        # need to map indices and slice into existing tensor
+        idxs = self._partition.map_to_global(
+            F.arange(0, tensor.shape[0], ctx=F.context(tensor)),
+            self._rank,
+        ).to(val.device)
+        val[idxs] = tensor.to(val.device)
+
+        self._store.delete_key(shared_name)
```

Comment:
Why delete it here? will it cause the `val` be revoked?

Reply:
> Why delete it here?

Say we try to save the checkpoint file at the end of every epoch, without deleting this key, from the second epoch and so on, processes with rank > 0 will try to get the shared array before rank 0 has created it, because the key is already in the TCPStore.

> Will it cause the val be revoked?

No. The object stored in `_store` is just a string.

Reply:
I see. It means each call will allocate a new address of memory. It should be ok as the local `state_dict` in pytorch should also return a new dict every time.